src/utility/Redact.ts

```diff
@@ -0,0 +1,25 @@
+// Please note: This utility function is written as short term usage.
+// Key aspect is to redact anything which we should log and is sensitive in nature.
+// Long term - we should aim for removing any data which we dont need and only need specific usage.
+enum SesitiveInfoFlags {
+    ClientSecret = "clientsecret",
```

Comment:
I know this is temporary fix before the long-term solution, can we also include/add an Enum for password and redact them as well? This is just a safety net for us because this extension produces lot of telemetry logs, and we might not if something else shows up in the feature. Thank you.

Reply:
☕️💡Yeah I did thought about it but if you look into  the json shared in workitems it’s specifically for client secret which is the field getting populated for the json. Essentially, it should be for every like: token, password, etc, but do we know any of these fields if so do let me know or update the workitem form privacy folks.

We should either do it on need to do basis aka following YAGNI or we donot for all possible things of known like token, creds, credentials, password.

This already little uncomfortable because it we be rather deleting these logs.

Reply:
I thought we can AVOID another work item from privacy in future by adding all possible fields/variables. If we want to deal it on need basis, I am fine with it. as a follow up we should refactor all these telemetry logs and keep only the one's that are used/needed. Thank you.

Reply:
here is issue for try/catch/throw patter: https://github.com/Azure/vscode-bridge-to-kubernetes/issues/31
here is issue for logging: https://github.com/Azure/vscode-bridge-to-kubernetes/issues/32

Please feel free to edit/comment and add more details.